### DISCUSSION TITLE
Search-based application with getStaticProps

### DISCUSSION THREAD

**Discussion by doptster** [vercel/next.js#22966] [Help] [answered]
I'm planning on building an NextJS search-based application, the requirement is we're going to need the server-side rendering for search results for SEO purposes.

In this scenario, is it better to use **getStaticProps** or **getInitialProps** for sever side rendering

**Comment by leerob** [comment, marked as answer]
Hey! For server-side rendering, you should use `getServerSideProps`. 

https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering
